src/web/editors/digitaljs.ts

```diff
@@ -45,6 +46,21 @@ export class DigitalJSEditor extends BaseEditor {
                 document: document.getText()
             });
         }
+        else if (message.type === 'requestSave') {
+            // TODO: figure out better way to save relative to project root
+            let rootPath = vscode.workspace.workspaceFolders?.[0].uri || vscode.Uri.file('.');
+            let path = vscode.Uri.joinPath(rootPath, message.data?.defaultPath || "");
```

Comment:
I enabled some stricter TypeScript rules, so `message.data` is probably of type `unknown` now. We can probably do some nice type definitions based on the `type` property, similar to `ExtensionMessage` in `src/messages.ts`

Reply:
I've added a new `ViewMessage` type to `src/extension/types.ts`.